### ISSUE TITLE
Echo Test Dials the Wrong Port

### ISSUE THREAD

**Issue by dle-baresip** [juha-h/baresip-studio#440]
I use GrapheneOS with baresip connecting to a personal FreePBX server hosted on a publicly accessible domain.  

Normal calling functions seem to work fine; baresip routes the calls correctly to my non-standard TLS port: 

> **TLS** 10.88.XX.XX:37782 -> 75.XX.XX.XX:**25061**
> INVITE sip:123@publicdomain.example.com:25061;transport=tls;user=phone SIP/2.0
> Via: SIP/2.0/TLS 10.88.XX.XX:37782;branch=z9hG4bKd3127110e70c2928;rport
> Contact: <sip:247-0x3932b21f10@10.88.XX.XX:37782;transport=tls>
> Max-Forwards: 70
> To: <sip:123@publicdomain.example.com:25061;transport=tls;user=phone>
> From: "Me" <sip:247@publicdomain.example.com:25061>;tag=dd8c76304a4fbbdb
> Call-ID: 522628a2609a093f
> CSeq: 41100 INVITE
> User-Agent: baresip v3.8.1 (?/?)
> Allow: INVITE,ACK,BYE,CANCEL,OPTIONS,NOTIFY,SUBSCRIBE,INFO,MESSAGE,UPDATE,REFER
> Supported: gruu,replaces,norefersub
> Content-Type: application/sdp
> Content-Length: 626

However, when I call out to one of the FreePBX special extensions (e.g. *43 echo test), baresip tries to open UDP 5060:

> **UDP** 10.88.XX.XX:37452 -> 75.164.XX.XX:**5060**
> INVITE sip:*43@publicdomain.example.com SIP/2.0
> Via: SIP/2.0/UDP 10.88.XX.XX:37452;branch=z9hG4bK38ea05ddc43cd3be;rport
> Contact: <sip:247-0x3932b21f10@10.88.XX.XX:37452>
> Max-Forwards: 70
> To: <sip:*43@publicdomain.example.com>
> From: "Me" <sip:247@publicdomain.example.com:25061>;tag=f1d2595d5ad62014
> Call-ID: a4e642197c86a15b
> CSeq: 24305 INVITE
> User-Agent: baresip v3.8.1 (?/?)
> Allow: INVITE,ACK,BYE,CANCEL,OPTIONS,NOTIFY,SUBSCRIBE,INFO,MESSAGE,UPDATE,REFER
> Supported: gruu,replaces,norefersub
> Content-Type: application/sdp
> Content-Length: 624

I have no plans to open port 5060 in my firewall.  I expected calls to *43 echo test to go to TLS 25061 all like the other calls.  Is my expectation wrong?

If it matters, this is how I configured my the baresip account port and protocol in the "Telephony Provider" field (as well as the account identity):

<img width="300" src="https://github.com/juha-h/baresip-studio/assets/110635346/108fbabd-8099-4cad-bf68-192d7ba78d7f">

**Comment by juha-h** [comment]
Most likely this is because baresip app does not recognize *43 as a telephone number and thus Telephony Provider is not used.  I would suggest that you add a contact for *43.  Otherwise there may be conflict of interest with other users of baresip that want *digits to be a normal URI user.

**Comment by dle-baresip** [comment]
Calling *43 works fine when baresip is running on VPN (i.e. port 5060 is open).  So there's no inherent limitation in either baresip or my PBX.  

Since you mentioned URI user, I made another test call to "blah".  In this case also baresip sends packets to UDP 5060.  So I think you are correct about the parsing distinction.  

Am I to understand that there are four separate models for baresip URI construction? 

* Registration: domain, port, and transport all specified at new account creation. 
* Telephony: domain, port, and transport all specified with the "Telephony Provider" account setting. 
* User name only: domain = registration domain, port = 5060, transport = UDP 
* Full URI: domain, port, and transport all specified on the dial screen

If true, I think that the "user name only" mode should be separately configurable, or follow the registration URI parameters.  I want to know that calls between baresip and my PBX are secure regardless of what kind of user baresip classifies them as.

**Comment by dle-baresip** [comment]
Also, I don't know whether this is intended behavior or not, but: 
* Calling "tel:*43" results in an error "Invalid SIP or tel URI sip:tel:*43".  
* Adding a contact for "*43" or "tel:*43" results in an error "Invalid SIP or tel URI tel:*43".  
* Calling or adding a fully qualified URI "tel:*43@publicdomain.example.com:25061;transport=tls" works.

**Comment by juha-h** [comment]
Character `*` is allowed in SIP URI userpart.  So it should work, if you add contact `Echo` with URI
```
sip:*43@publicdomain.example.com:25061;transport=tls
```

**Comment by juha-h** [comment]
If you you have Telephony Provider defined for an account, baresip uses it for calls to telephone numbers,  *43 is not telephone number and thus Telephony Provider is not used.   Instead, calling *43 is the same calling *43@<domain of your account>.  If your account has Outbound Proxy defined, the call is routed there.  If not, the call is routed based on NAPTR/SRV/A lookup of <domain of your account>.

**Comment by dle-baresip** [comment]
Thanks, I think that's enough to solve my problem.  First, I am aware of the NAPTR/SRV records, but I didn't think those are suitable solutions in this case because they don't support alternative ports.  Plus, even if DNS records did support alternative ports I would rather not have public signposts to my private PBX.  

Setting the Outbound Proxy the same as the Telephony Provider was the solution.  After that change, dialing a plain `*43` routes the call on the port / protocol that I want.  

I guess I made two wrong assumptions along the way:  
* First, I assumed that when I specified the `domain:port;transport` at account creation that all three of those factors would be used for all communications unless overridden.  Instead, it appears that `domain:port;transport` only apply to registration, while other communications only use `domain`.
* Second, I assumed that the Outbound Proxy was only relevant for `A -> Proxy -> B` type routes -- similar to the proxies used for web browsing -- where `A` and `B` could not otherwise communicate.  It did not occur to me that this configuration field could be co-opted to redefine the end destination `B`.

Anyway, to correct the table from earlier:
*    Registration: domain, port, and transport defined at new account creation.
*    Telephony: domain, port, and transport defined with the "Telephony Provider" account setting.
*    User name only: **domain, port, and transport defined with the "Outbound Proxy" account setting.**
*    Full URI: domain, port, and transport all specified on the dial screen<sup>(1)</sup>.

Thanks again.

(1) Although, when Outbound Proxy is set, calls to `user@thirdpartydomain.example.com` also go through the proxy.   Not a concern for my current use case where baresip acts as a remote extension, but I can imagine a scenario where someone might object to the extra hop.
